src/hffs/fs.py

```diff
@@ -18,7 +19,17 @@
 
 
 PY_VERSION = version.parse(platform.python_version())
-HFH_VERSION = version.parse(huggingface_hub.__version__)
+RE_NEXT_LINK = re.compile(r'<([^>]+)>; rel="next"')
+
+if PY_VERSION >= version.parse("3.11"):
+
+    def _get_tz_aware_date_from_iso_date_str(date: str) -> datetime:
+        return datetime.fromisoformat(date)
+
+else:
+
+    def _get_tz_aware_date_from_iso_date_str(date: str) -> datetime:
+        return datetime.fromisoformat(date.rstrip("Z")).replace(tzinfo=timezone.utc)
```

Comment:
There is a "undocumented yet public" helper to parse a datetime returned by the server as a python object:  [`from huggingface_hub.utils import parse_datetime`](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/utils/_datetime.py#LL25C5-L25C19).

The implementation is not exactly the same (especially no distinction between 3.x and 3.11 versions) so could you try that you get exactly the same result as what I did? If not, it would be nice to open an issue on `huggingface_hub` side to fix it. And in both cases, you can use the one from `huggingface_hub` to avoid duplication.

(sorry if that type of helper is not really documented, it's quite internal to be honest)